### PR TITLE
Calculate form return_url for post and message instead of using SESSION Variable

### PR DESCRIPTION
**Pull request by JonnyTischbein**
Resubmit pull request for #3897 .
I calculate in `mod/message` and `mod/edipost` the correct return_url. 
Since there are many places from which you can edit a post I added the return_url as a GET parameter in the function call encoded in bas64 because the curl could contain subdirs like in `/profile/[user]`

### CODE REVIEW

**Comment by JonnyTischbein** [comment]
Yes,
I tested editing different Posts from different locations and sending Messages to different user.
Everytime I was redirected correctly.

**Comment by JonnyTischbein** [comment]
Only downside: If you edit your post at the bottom of your network page you are redirected to the top.

I have tried to redirect to `/network#item-91bfdaa4225ba27b394212c557968837` but it is not the right position because of the header menus.

**Comment by MrPetovan** [comment]
Ideally the edition would happen in a modal window whose data could be sent via an Ajax call without triggering a page reload, but this is a good improvement already.

File: mod/editpost.php [hunk 1]
```diff
@@ -21,12 +21,18 @@ function editpost_content(App $a)
...
+	$return_url = (($a->argc > 2) ? base64_decode($a->argv[2]) : '');
 
 	if (!$post_id) {
 		notice(L10n::t('Item not found') . EOL);
 		return;
 	}
 
+	// Fallback to SESSION return_path
+	if (empty($return_url)) {
+		$return_url = $_SESSION['return_path'];
```
Comment (MrPetovan) [inline mod/editpost.php]: Isn't this supposed to be `$_SESSION['return_url']`?

Comment (JonnyTischbein) [reply mod/editpost.php]: It is! Sorry for that..

Comment (MrPetovan) [reply mod/editpost.php]: No problem, you can add a new commit in your branch and submit a new PR.

**Pull request merged**